configure.ac

```diff
@@ -0,0 +1,315 @@
+	echo '   $1' >& AS_MESSAGE_FD
+	echo '   $1   ' | sed -e's/./-/g' >&AS_MESSAGE_FD
+])
+
+dnl works for checking define statements in c
+m4_include([m4/AX_CHECK_DEFINE.m4])
+
+freebsd_src=${FREEBSD_SRC:=/usr/src}
+
+dnl Setting appropriate CFLAGS
+
+CFLAGS="-nostdinc 
+-I${freebsd_src}/cddl/lib/libumem 
+-I${freebsd_src}/sys/cddl/compat/opensolaris/ 
+-I${freebsd_src}/sys/cddl/compat/opensolaris 
+-I${freebsd_src}/cddl/compat/opensolaris/include 
+-I${freebsd_src}/cddl/compat/opensolaris/lib/libumem 
+-I${freebsd_src}/cddl/contrib/opensolaris/lib/libzpool/common 
+-I${freebsd_src}/sys/cddl/contrib/opensolaris/common/zfs 
+-I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common/fs/zfs 
+-I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common/sys 
+-I${freebsd_src}/cddl/contrib/opensolaris/head 
+-I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common 
+-I${freebsd_src}/cddl/contrib/opensolaris/lib/libnvpair 
+-I${freebsd_src}/cddl/contrib/opensolaris/lib/libuutil/common 
+-I${freebsd_src}/cddl/contrib/opensolaris/lib/libzfs/common 
+-I${freebsd_src}/cddl/contrib/opensolaris/lib/libzfs_core/common 
+${CFLAGS} -I/usr/include"
+
+dnl Setting appropriate LDFLAGS
+LDFLAGS="-lzfs -lnvpair -lzfs_core -luutil -lgeom -v"
+
+
+AC_DEFINE(NEED_SOLARIS_BOOLEAN)
+AC_DEFINE(_XPG6)
+
+
+SECTION_TITLE([RUNNING TESTS ON zfs.h HEADER])
+
+AC_CHECK_HEADERS([sys/types.h], [], [AC_ERROR([A working sys/types header is required])])
+AC_CHECK_HEADERS([sys/fs/zfs.h], [], [AC_ERROR([A working zfs header is required])],[
+	#include <sys/types.h>
+])
+
+dnl following is a valid check if zfs_max_dataset_name_len exists
+AX_CHECK_DEFINE(
+	[sys/fs/zfs.h], [ZFS_MAX_DATASET_NAME_LEN],
+	[AC_SUBST(HAVE_ZFS_MAX_DATASET_NAME_LEN, 1)],
+	[AC_SUBST(HAVE_ZFS_MAX_DATASET_NAME_LEN, 0)]
+)
+
+
+dnl Checking existence of pss_pass_scrub_pause & pss_pass_scrub_spent_paused struct member
+AC_MSG_NOTICE([Checking existence of pss_pass_scrub_pause & pss_pass_scrub_spent_paused struct member])
+
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+		[#include <sys/fs/zfs.h>],
+		[
+			pool_scan_stat_t o;
+			o.pss_pass_scrub_pause = 1;
+			o.pss_pass_scrub_spent_paused = 1 /* ; */
+		]
+	)],
+	[AC_SUBST(HAVE_POOL_SCAN_STAT_T_PAUSE, 1) AC_MSG_NOTICE([OK])],
+	[AC_SUBST(HAVE_POOL_SCAN_STAT_T_PAUSE, 0) AC_MSG_NOTICE([Not found])]
+)
+
+
+dnl Checking typedef pool_scrub_cmd_t in zfs header
+AC_MSG_NOTICE([Checking typedef pool_scrub_cmt_t in zfs header])
+
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+		[#include <sys/fs/zfs.h>],
+		[
+			int i = sizeof(pool_scrub_cmd_t);
+			i = i; /* ; */
+		]
+	)],
+	[AC_SUBST(HAVE_POOL_SCRUB_CMD_T, 1) AC_MSG_NOTICE([OK])],
+	[AC_SUBST(HAVE_POOL_SCRUB_CMD_T, 0) AC_MSG_NOTICE([Not found])]
+)
+
+
+SECTION_TITLE([RUNNING TESTS ON libzfs.h HEADER])
+
+
+AC_CHECK_HEADERS([libzfs.h], [], [AC_ERROR([A working libzfs header is required])], [
+	#include <sys/types.h>
+	#include <sys/mnttab.h>
+	#include <sys/fs/zfs.h>
+	#include <sys/zfs_ioctl.h>
```

Comment:
can you give us a quick run-down on which features the `sys/zfs_ioctl` header is used for?

Reply:
When checking for `libzfs.h` header, `sys/zfs_ioctl` is a header which is being used by `libzfs.h` and thus we include it as a dependency.
`#include <sys/zfs_ioctl.h>` Taken from `libzfs.h`

Reply:
aye! thanks for the explanation!

Reply:
No problem